scrapi/harvesters/springer.py

```diff
@@ -0,0 +1,153 @@
+"""
+BioMed Central harvester of public projects for the SHARE Notification Service
+Note: At the moment, this harvester only harvests basic data on each article, and does
+not make a seperate request for additional metadata for each record.
+
+Example API query: http://api.springer.com/meta/v1/json?api_key=yourapikey&q=*&s=1&p=200
+"""
+
+from __future__ import unicode_literals
+
+import json
+import logging
+from furl import furl
+from datetime import date, timedelta
+
+from nameparser import HumanName
+
+from scrapi import requests
+from scrapi import settings
+from scrapi.base import JSONHarvester
+from scrapi.linter.document import RawDocument
+from scrapi.base.helpers import build_properties, date_formatter
+
+logger = logging.getLogger(__name__)
+
+
+def process_contributors(authors):
+
+    contributor_list = []
+    for person in authors:
+        name = HumanName(person['creator'])
+        contributor = {
+            'name': person['creator'],
+            'givenName': name.first,
+            'additionalName': name.middle,
+            'familyName': name.last
+        }
+        contributor_list.append(contributor)
+
+    return contributor_list
+
+
+def process_urls(urls):
+    all_uris = [url['value'] for url in urls]
+
+    processed_uris = {
+        'canonicalUri': all_uris[0],
+        'objectUris': [],
+        'providerUris': all_uris
+    }
+
+    for url in all_uris:
+        if 'dx.doi.org' in url:
+            processed_uris['objectUris'].append(url)
+
+    return processed_uris
+
+
+class SpringerlHarvester(JSONHarvester):
+    short_name = 'springer'
+    long_name = 'Springer'
+    url = 'http://www.springer.com/us/'
+    count = 0
+
+    base_url = 'http://api.springer.com/meta/v1/json'
+    arged_url = furl(base_url)
+    arged_url.args['api_key'] = settings.SPRINGER_KEY
+    arged_url.args['p'] = 100
+
+    URL = arged_url.url
+
+    @property
+    def schema(self):
+        return {
+            'contributors': ('/creators', process_contributors),
+            'uris': ('/url', process_urls),
+            'title': '/title',
+            'providerUpdatedDateTime': ('/publicationDate', date_formatter),
+            'description': '/abstract',
+            'freeToRead': {
+                'startDate': ('/openaccess', '/publicationDate', lambda x, y: y if x == 'true' else None)
+            },
+            'publisher': {
+                'name': '/publisher'
+            },
+            'otherProperties': build_properties(
+                ('url', '/url'),
```

Comment:
Reason I included it twice was because urls in the original also includes inner format and platform that are interesting in context, but I wasn't sure how to include in the URL field

```
url: [
    {
        format: "html",
        platform: "web",
        value: "http://link.springer.com/openurl/fulltext?id=doi:10.1007/978-3-642-54596-2_201165"
    },
    {
       format: "",
       platform: "",
       value: "http://dx.doi.org/10.1007/978-3-642-54596-2_201165"
    }
]
```